lib/internal/file.js

```diff
@@ -0,0 +1,117 @@
+'use strict';
+
+const {
+  DateNow,
+  NumberIsNaN,
+  ObjectDefineProperties,
+  ObjectPrototypeHasOwnProperty,
+  Symbol,
+  SymbolToStringTag,
+} = primordials;
+
+const {
+  Blob,
+} = require('internal/blob');
+
+const {
+  customInspectSymbol: kInspect,
+  emitExperimentalWarning,
+  kEnumerableProperty,
+  kEmptyObject,
+  toUSVString,
+} = require('internal/util');
+
+const {
+  codes: {
+    ERR_INVALID_THIS,
+    ERR_MISSING_ARGS,
+  },
+} = require('internal/errors');
+
+const {
+  inspect,
+} = require('internal/util/inspect');
+
+const kState = Symbol('kState');
+
+function isFile(object) {
+  return object?.[kState] !== undefined || object instanceof Blob;
+}
+
+class File extends Blob {
+  constructor(fileBits, fileName, options = kEmptyObject) {
+    emitExperimentalWarning('buffer.File');
+
+    if (arguments.length < 2) {
+      throw new ERR_MISSING_ARGS('fileBits', 'fileName');
+    }
+
+    super(fileBits, options);
+
+    let lastModified;
+
+    if (ObjectPrototypeHasOwnProperty(options, 'lastModified')) {
+      // Using Number(...) will not throw an error for bigints.
+      lastModified = +options.lastModified;
+
+      if (NumberIsNaN(lastModified)) {
+        lastModified = 0;
+      }
+    } else {
+      lastModified = DateNow();
+    }
+
+    this[kState] = {
+      name: toUSVString(fileName),
+      lastModified,
+    };
+  }
+
+  get name() {
+    if (!isFile(this)) {
+      throw new ERR_INVALID_THIS('File');
+    }
```

Comment:
Is it correct to accept `Blob` instances here?

```console
> Object.getOwnPropertyDescriptor(File.prototype, 'name').get.call(new Blob)
TypeError: 'get name' called on an object that does not implement interface File.
```

Reply:
no, see comment above